### ISSUE TITLE
final_target_logit = torch.where(target_logit > self.threshold, cos_theta_m, target_logit - self.mm)

### ISSUE THREAD

**Issue by powermano** [HuangYG123/CurricularFace#31]
In MV-softmax, the released code use easy_margin like following:
```
final_gt = torch.where(gt > 0, cos_theta_m, gt)
```
but the official arcface implement does not use easy_margin, like following:
```
final_gt =  cos(θ + m)
```
I try mv-softmax using own dataset without easy_margin, but suffering from divergence issue(NAN). I fix divergence issue using easy_margin.

Compared with mv_softmax , there are two diffs, first is the thresh, CurricularFace use ```self.threshold = math.cos(math.pi - m)```,  this is understandable， second is ``` final_target_logit = torch.where(target_logit > self.threshold, cos_theta_m, target_logit - self.mm)```,  what is intuitive understanding of ```target_logit - self.mm ```, where ```self.mm = math.sin(math.pi - m) * m  ```

**Comment by powermano** [comment]
My understanding is that , the arc length of margin radian degrees is calculated as ``` L = mr ```  where ``` r = 1```,  so the ```L = m ```,  so the approximate calculation of 
```
final_gt =  cos(θ + m)
```
is 
```
final_gt = cos(θ) - L*sin(m) =  cos(θ) - m * sim(m) =  cos(θ) - sim(math.pi - m) * m
```

Is above right? if right, what is the advantages of this implements.

**Comment by lizhenstat** [comment]
Hi, same question here, did you find the meaning of self.mm later? Thanks a lot. @powermano
